### PR TITLE
Make game log more efficient

### LINKED ISSUE #12: Make game log more efficient
The player-facing game log stores the 200 most recent lines in the game state. While this is convenient to access, this is memory inefficient, since each line of the log will be redundantly copied in several game states. We need a new implementation for the game log that stores each line of the log only once.

One possibility would be to store only the new lines of the log to the game state. When it comes time to render the game log, the client would go through the previous game states and pull those log lines.

### PR DESCRIPTION
**Pull request by kevinddchen**
The game log is no longer saved in the game state `G` but is now saved in the built-in game log, and is accessed by the `Board` component. This resolves #12.

- Game log no longer limited to ~200 lines
- Game log is slightly more verbose
- Less load on latency, since the entire game log no longer needs to be passed between the server and client.

### CODE REVIEW

**Comment by kvnhykw** [comment]
If I touch anything, it will break. I’m not even sure what the ramifications of posting this comment is.

**Comment by kevinddchen** [comment]
Thou must approve this message

**Review by AquaDragon — APPROVED** [review]
looks good

File: src/board/Logger.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+      case LogEvent.RollOne: {
+        return `\troll ${logLine.roll}`;
+      }
+      case LogEvent.RollTwo: {
+        const { dice } = logLine;
+        const roll = dice[0] + dice[1];
+        return `\troll ${roll} (${dice})`;
+      }
+      case LogEvent.AddTwo: {
+        return `\tchange roll to ${logLine.roll}`;
```
Comment (AquaDragon) [inline src/board/Logger.tsx:73]: change -> changes

File: src/board/Logger.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    const { event } = logLine;
+
+    switch (event) {
+      case LogEvent.RollOne: {
+        return `\troll ${logLine.roll}`;
+      }
+      case LogEvent.RollTwo: {
+        const { dice } = logLine;
+        const roll = dice[0] + dice[1];
+        return `\troll ${roll} (${dice})`;
```
Comment (AquaDragon) [inline src/board/Logger.tsx:70]: line 65/70: roll -> rolls

File: src/board/Logger.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+   * @param logLine
+   * @returns
+   */
+  parseLogLine = (logLine: LogLine): string | null => {
+    const { names } = this.props;
+    const { event } = logLine;
+
+    switch (event) {
+      case LogEvent.RollOne: {
+        return `\troll ${logLine.roll}`;
```
Comment (AquaDragon) [inline src/board/Logger.tsx:65]: we should make the tenses the same for consistency. past tense maybe?
line 65/70: roll -> rolled
73: change -> changed
77: earns -> earned
81: pays -> paid
84: buy -> bought
88: trade -> traded

Comment (kevinddchen) [reply src/board/Logger.tsx:65]: sure, good suggestion. I'll make these changes.

**Pull request merged**